### ISSUE TITLE
Implicit dependency on npm

### ISSUE THREAD

**Issue by nkbt** [timoxley/npm-which#6]
When using `mhart/alpine-node:base-7.2.0` base image for docker container, that has only Node without npm this happens: 

```
/usr/src/app/node_modules/npm-which/index.js:58
      if (err) throw err
               ^

Error: not found: npm
    at getNotFoundError (/usr/src/app/node_modules/which/which.js:13:12)
    at Function.whichSync [as sync] (/usr/src/app/node_modules/which/which.js:131:9)
    at whichNpm (/usr/src/app/node_modules/npm-path/index.js:150:23)
    at getPath (/usr/src/app/node_modules/npm-path/index.js:26:3)
    at Function.getPathSync [as getSync] (/usr/src/app/node_modules/npm-path/index.js:74:3)
    at Function.npmWhich.sync (/usr/src/app/node_modules/npm-which/index.js:50:29)
    at new exports.StatsdSubmit (/usr/src/app/node_modules/@packetloop/statsd-submit/lib/statsdSubmit.js:13:31)
    at Object.<anonymous> (/usr/src/app/node_modules/@packetloop/log/lib/requestLog.js:27:16)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
```

If library depends on NPM then it should probably be in dependencies.

README update would be neat too

Cheers 🍻

**Comment by nkbt** [comment]
Sorry this is a `which` issue.

**Comment by nkbt** [comment]
Sorry, this is apparently `npm-path` problem.

**Comment by timoxley** [comment]
It's been a while since I've looked at these packages, can't recall how/why it needs npm… 

Though I do recall in the past, perhaps for another package being a bit reluctant to add all the weight of an npm dependency, having node but no npm is a bit of an edge case and an npm dependency would bloat this package up in a big way.

**Comment by nkbt** [comment]
Yeah, that is partially my point. I used npm-which to actually not use npm,
as lightweight alternative, though it actually used npm internally and that
sort of devalues this idea.

**Comment by nkbt** [comment]
And on a side note, alpine node image without npm shaves off extra ~15mb
from already tiny docker container (about -30%). Npm is almost never needed
for production build, so it is not that edge case really :)

**Comment by timoxley** [comment]
note that `npm-path` doesn't actually *boot* `npm`. in order to accurately emulate what `npm` itself does to the `PATH`, `npm-path` simply uses the location of `npm` to append the appropriate location of `node-gyp-bin` to the `PATH`.

https://github.com/timoxley/npm-path/blob/f9248253470abd003d3898a8b7b69ede3426d30d/index.js#L36-L37

**Comment by timoxley** [comment]
perhaps this should just silently fail and move on if `npm` is not found

**Comment by nkbt** [comment]
Makes sense. That would be nice 👍

**Comment by timoxley** [comment]
@nkbt Released in [npm-path@2.0.3](https://github.com/timoxley/npm-path/compare/2.0.2...2.0.3)

**Comment by nkbt** [comment]
Perfect, thank you a lot!

Here is a diff for two docker builds of our app with and without npm. Last one does not fail anymore with your update!
```
IMAGE ID                              CREATED              SIZE
[mhart/alpine-node:base-7.2.0]        About a minute ago   68.27 MB
[mhart/alpine-node:7.2.0]             3 hours ago          80.55 MB
```
